Various configuration files.

Most notibly, the cached results of

./configure --prefix=/usr --libdir=/usr/lib64 --libexecdir=/usr/libexec --disable-seabios --disable-stubdom --disable-xsmpolicy --disable-blktap2 --with-system-qemu=/usr/lib64/xen/bin/qemu-system-i386 --with-xenstored=oxenstored --enable-systemd

diff --git a/.config b/.config
new file mode 100644
index 0000000..44b7e63
--- /dev/null
+++ b/.config
@@ -0,0 +1,13 @@
+# -*- mode: Makefile; -*-
+
+# Perform a debug build for trunk
+debug := y
+
+# Always produce debug symbols for debuginfo RPMs
+debug_symbols := y
+
+# No tests
+CONFIG_TESTS := n
+
+# No remus netbuffering
+override CONFIG_REMUS_NETBUF := n
diff --git a/config/Docs.mk b/config/Docs.mk
new file mode 100644
index 0000000..b1a32cb
--- /dev/null
+++ b/config/Docs.mk
@@ -0,0 +1,10 @@
+-include $(XEN_ROOT)/config/Paths.mk
+
+# Tools
+FIG2DEV             := 
+POD2MAN             := /usr/bin/pod2man
+POD2HTML            := /usr/bin/pod2html
+POD2TEXT            := /usr/bin/pod2text
+MARKDOWN            := 
+PANDOC              := 
+PERL                := /usr/bin/perl
diff --git a/config/Paths.mk b/config/Paths.mk
new file mode 100644
index 0000000..dc97bc1
--- /dev/null
+++ b/config/Paths.mk
@@ -0,0 +1,63 @@
+# Xen system configuration
+# ========================
+#
+# Xen uses a set of variables for system configuration and at build time,
+# because of this these variables are defined on one master input source file
+# and is generated after running ./configure. The master source is located
+# on the xen source tree at under config/Paths.mk.in and it is used to
+# generate shell or header files by the build system upon demand through the
+# use of the helper makefile helper buildmakevars2file().
+#
+# For more documentation you can refer to the wiki:
+#
+# http://wiki.xen.org/wiki/Category:Host_Configuration#System_wide_xen_configuration
+
+PACKAGE_TARNAME          := xen
+prefix                   := /usr
+bindir                   := /usr/bin
+sbindir                  := /usr/sbin
+libdir                   := /usr/lib64
+libexecdir               := /usr/libexec
+datarootdir              := ${prefix}/share
+mandir                   := ${datarootdir}/man
+docdir                   := ${datarootdir}/doc/${PACKAGE_TARNAME}
+dvidir                   := ${docdir}
+htmldir                  := ${docdir}
+pdfdir                   := ${docdir}
+psdir                    := ${docdir}
+includedir               := ${prefix}/include
+localstatedir            := /var
+sysconfdir               := /etc
+
+PREFIX                   := $(prefix)
+
+SBINDIR                  := $(sbindir)
+BINDIR                   := $(bindir)
+LIBEXEC                  := $(libexecdir)/$(PACKAGE_TARNAME)
+LIBEXEC_BIN              := /usr/libexec/xen/bin
+LIBEXEC_LIB              := $(LIBEXEC)/lib
+LIBEXEC_INC              := $(LIBEXEC)/include
+
+INCLUDEDIR               := $(includedir)
+SHAREDIR                 := /usr/share
+MANDIR                   := $(mandir)
+MAN1DIR                  := $(MANDIR)/man1
+MAN8DIR                  := $(MANDIR)/man8
+LIBDIR                   := $(libdir)
+DOCDIR                   := $(docdir)
+
+XEN_RUN_DIR              := /var/run/xen
+XEN_LOG_DIR              := /var/log/xen
+XEN_LIB_STORED           := /var/lib/xenstored
+
+CONFIG_DIR               := /etc
+INITD_DIR                := /etc/rc.d/init.d
+CONFIG_LEAF_DIR          := sysconfig
+BASH_COMPLETION_DIR      := $(CONFIG_DIR)/bash_completion.d
+XEN_LOCK_DIR             := /var/lock
+XEN_PAGING_DIR           := /var/lib/xen/xenpaging
+
+XENFIRMWAREDIR           := $(LIBEXEC)/boot
+
+XEN_CONFIG_DIR           := /etc/xen
+XEN_SCRIPT_DIR           := /etc/xen/scripts
diff --git a/config/Stubdom.mk b/config/Stubdom.mk
new file mode 100644
index 0000000..a8bb1d3
--- /dev/null
+++ b/config/Stubdom.mk
@@ -0,0 +1,45 @@
+# Prefix and install folder
+prefix              := /usr/local
+PREFIX              := $(prefix)
+exec_prefix         := ${prefix}
+libdir              := ${exec_prefix}/lib
+LIBDIR              := $(libdir)
+
+# Path Programs
+CMAKE               := /usr/bin/cmake
+FETCHER             := /usr/bin/wget -c -O
+
+# A debug build of stubdom? //FIXME: Someone make this do something
+debug               := y
+vtpm = y
+
+STUBDOM_TARGETS     :=  ioemu grub xenstore vtpm vtpmmgr
+STUBDOM_BUILD       :=  ioemu-stubdom pv-grub xenstore-stubdom vtpm-stubdom vtpmmgr-stubdom
+STUBDOM_INSTALL     :=  install-ioemu install-grub install-xenstore install-vtpm install-vtpmmgr
+
+ZLIB_VERSION        := 1.2.3
+ZLIB_URL            := $(XEN_EXTFILES_URL)
+
+LIBPCI_VERSION      := 2.2.9
+LIBPCI_URL          := $(XEN_EXTFILES_URL)
+
+NEWLIB_VERSION      := 1.16.0
+NEWLIB_URL          := $(XEN_EXTFILES_URL)
+
+LWIP_VERSION        := 1.3.0
+LWIP_URL            := $(XEN_EXTFILES_URL)
+
+GRUB_VERSION        := 0.97
+GRUB_URL            := $(XEN_EXTFILES_URL)
+
+OCAML_VERSION       := 3.11.0
+OCAML_URL           := http://caml.inria.fr/pub/distrib/ocaml-3.11
+
+GMP_VERSION         := 4.3.2
+GMP_URL             := $(XEN_EXTFILES_URL)
+
+POLARSSL_VERSION    := 1.1.4
+POLARSSL_URL        := $(XEN_EXTFILES_URL)
+
+TPMEMU_VERSION      := 0.7.4
+TPMEMU_URL          := $(XEN_EXTFILES_URL)
diff --git a/config/Tools.mk b/config/Tools.mk
new file mode 100644
index 0000000..2c38670
--- /dev/null
+++ b/config/Tools.mk
@@ -0,0 +1,80 @@
+-include $(XEN_ROOT)/config/Paths.mk
+
+CONFIG_RUMP         := n
+ifeq ($(CONFIG_RUMP),y)
+XEN_OS              := NetBSDRump
+endif
+
+# A debug build of tools?
+debug               := y
+
+# Tools path
+BISON               := /usr/bin/bison
+FLEX                := /usr/bin/flex
+PYTHON              := python
+PYTHON_PATH         := /usr/bin/python
+PERL                := /usr/bin/perl
+CURL_CONFIG         := 
+XML2_CONFIG         := 
+BASH                := /bin/sh
+XGETTTEXT           := /usr/bin/xgettext
+AS86                := /usr/bin/as86
+LD86                := /usr/bin/ld86
+BCC                 := /usr/bin/bcc
+IASL                := /usr/bin/iasl
+FETCHER             := /usr/bin/wget -c -O
+SEABIOS_PATH        := 
+OVMF_PATH           := 
+
+# Extra folder for libs/includes
+PREPEND_INCLUDES    := 
+PREPEND_LIB         := 
+APPEND_INCLUDES     := 
+APPEND_LIB          := 
+
+PTHREAD_CFLAGS      := -pthread
+PTHREAD_LDFLAGS     := -pthread
+PTHREAD_LIBS        := 
+
+PTYFUNCS_LIBS       := -lutil
+
+LIBNL3_LIBS         := 
+LIBNL3_CFLAGS       := 
+XEN_TOOLS_RPATH     := n
+
+# Download GIT repositories via HTTP or GIT's own protocol?
+# GIT's protocol is faster and more robust, when it works at all (firewalls
+# may block it). We make it the default, but if your GIT repository downloads
+# fail or hang, please specify GIT_HTTP=y in your environment.
+GIT_HTTP            := n
+
+# Optional components
+XENSTAT_XENTOP      := y
+OCAML_TOOLS         := y
+FLASK_POLICY        := n
+CONFIG_OVMF         := n
+CONFIG_ROMBIOS      := y
+CONFIG_SEABIOS      := n
+CONFIG_QEMU_TRAD    := y
+CONFIG_QEMU_XEN     := n
+CONFIG_BLKTAP1      := n
+CONFIG_BLKTAP2      := n
+CONFIG_QEMUU_EXTRA_ARGS:= 
+CONFIG_REMUS_NETBUF := n
+
+CONFIG_SYSTEMD      := y
+SYSTEMD_CFLAGS      :=  
+SYSTEMD_LIBS        := -lsystemd-daemon  
+XEN_SYSTEMD_DIR     := $(prefix)/lib/systemd/system/
+XEN_SYSTEMD_MODULES_LOAD := $(prefix)/lib/modules-load.d/
+
+LINUX_BACKEND_MODULES := xen-evtchn xen-gntdev xen-gntalloc xen-blkback xen-netback xen-pciback evtchn gntdev netbk blkbk xen-scsibk usbbk pciback xen-acpi-processor blktap2 blktap
+
+#System options
+ZLIB                :=  -DHAVE_BZLIB -lbz2 -DHAVE_LZMA -llzma
+CONFIG_LIBICONV     := n
+CONFIG_GCRYPT       := y
+EXTFS_LIBS          := -lext2fs
+CURSES_LIBS         := -lncurses
+
+FILE_OFFSET_BITS    := 
diff --git a/config/Toplevel.mk b/config/Toplevel.mk
new file mode 100644
index 0000000..ced7d97
--- /dev/null
+++ b/config/Toplevel.mk
@@ -0,0 +1 @@
+SUBSYSTEMS               :=  xen tools docs
diff --git a/tools/config.h b/tools/config.h
new file mode 100644
index 0000000..0ebc9df
--- /dev/null
+++ b/tools/config.h
@@ -0,0 +1,112 @@
+/* config.h.  Generated from config.h.in by configure.  */
+/* config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* Blktap2 enabled */
+/* #undef HAVE_BLKTAP2 */
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the `aio' library (-laio). */
+/* #undef HAVE_LIBAIO */
+
+/* Define to 1 if you have the `crypto' library (-lcrypto). */
+#define HAVE_LIBCRYPTO 1
+
+/* Define to 1 if you have the `fdt' library (-lfdt). */
+/* #undef HAVE_LIBFDT */
+
+/* Define to 1 if you have the `yajl' library (-lyajl). */
+#define HAVE_LIBYAJL 1
+
+/* Define to 1 if you have the `z' library (-lz). */
+#define HAVE_LIBZ 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#define HAVE_MEMORY_H 1
+
+/* Qemu traditional enabled */
+#define HAVE_QEMU_TRADITIONAL 1
+
+/* ROMBIOS enabled */
+#define HAVE_ROMBIOS 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#define HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#define HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Systemd available and enabled */
+#define HAVE_SYSTEMD 1
+
+/* Define to 1 if you have the <sys/eventfd.h> header file. */
+#define HAVE_SYS_EVENTFD_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#define HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the <utmp.h> header file. */
+#define HAVE_UTMP_H 1
+
+/* Define to 1 if you have the <valgrind/memcheck.h> header file. */
+/* #undef HAVE_VALGRIND_MEMCHECK_H */
+
+/* Define to 1 if you have the <yajl/yajl_version.h> header file. */
+#define HAVE_YAJL_YAJL_VERSION_H 1
+
+/* Define curses header to use */
+#define INCLUDE_CURSES_H <ncurses.h>
+
+/* Define extfs header to use */
+#define INCLUDE_EXTFS_H <ext2fs/ext2fs.h>
+
+/* libutil header file name */
+/* #undef INCLUDE_LIBUTIL_H */
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "xen-devel@lists.xen.org"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Xen Hypervisor Tools"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "Xen Hypervisor Tools 4.5"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "xen"
+
+/* Define to the home page for this package. */
+#define PACKAGE_URL "http://www.xen.org/"
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "4.5"
+
+/* Qemu Xen path */
+#define QEMU_XEN_PATH "/usr/lib64/xen/bin/qemu-system-i386"
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* Enable large inode numbers on Mac OS X 10.5.  */
+#ifndef _DARWIN_USE_64_BIT_INODE
+# define _DARWIN_USE_64_BIT_INODE 1
+#endif
+
+/* Number of bits in a file offset, on hosts where this is settable. */
+/* #undef _FILE_OFFSET_BITS */
+
+/* Define for large files, on AIX-style hosts. */
+/* #undef _LARGE_FILES */
